File: in_Java/0145 Binary Tree Postorder Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        if(root == null) return res;
        helper(res,root);
        return res;
    }
    private void helper(List<Integer> res, TreeNode root){
        if(root == null) return;
        helper(res,root.left);
        helper(res,root.right);
        res.add(root.val);
    }
}

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        if(root == null) return res;
        Stack<TreeNode> st1 = new Stack<>();
        Stack<TreeNode> st2 = new Stack<>();
        st1.push(root);
        while(!st1.isEmpty()){
            TreeNode cur = st1.pop();
            st2.push(cur);
            if(cur.left != null) st1.push(cur.left);
            if(cur.right != null) st1.push(cur.right);
        }
        while(!st2.isEmpty()) res.add(st2.pop().val);
        return res;
    }
}

File: in_Python/0145 Binary Tree Postorder Traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def postorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        res = []
        st = []
        if root == None:
            return res
        st.append(root)
        while len(st) != 0:
            h = st.pop()
            if h.left != None:
                st.append(h.left)
            if h.right != None:
                st.append(h.right)
            res.insert(0,h.val)
        return res